#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QHash>
#include <QPair>

#include "infoeleitor.h"

namespace Ui {
class MainWindow;
}

enum State {
    WAIT_DIG_1,
    WAIT_DIG_2,
    SHOW_CAND,
    END,
};

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    explicit MainWindow(QWidget *parent = 0);
    ~MainWindow();
protected:
    void keyPressEvent(QKeyEvent* event);
    void closeEvent(QCloseEvent *event);
private slots:
    void soundPlayingChanged();
private:
    void visibilidadeInicial();
    void visibilidadeCandidato();
    void visibilidadeNulo();
    void visibilidadeFim();
    void contabilizaCedula(Cedula &cedula);
    void boletimUrna(const char *titulo = "BOLETIM DE URNA");

    Ui::MainWindow *ui;
    State state;
    QHash<QString, QPair<QString, QString>> candidatos;
    QHash<QString, int> votos_candidatos;
    int votos_nulos = 0;
};

#endif // MAINWINDOW_H
